### PR TITLE
Audit all fields for finality.

### PR DESCRIPTION
**Pull request by AdamBJohnsonx**
_No description provided._

---
Edit by @iambmelt:
Related crash report:
```
com.microsoft.identity.common.internal.controllers.CommandDispatcher$1.run CommandDispatcher.java:151
java.util.concurrent.ThreadPoolExecutor.runWorker ThreadPoolExecutor.java:1167
java.util.concurrent.ThreadPoolExecutor$Worker.run ThreadPoolExecutor.java:641
java.lang.Thread.run Thread.java:919
```

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/LocalMSALController.java [hunk 1]
```diff
@@ -83,9 +83,11 @@
     private static final String TAG = LocalMSALController.class.getSimpleName();
 
     @SuppressWarnings(WarningType.rawtype_warning)
+    @EqualsAndHashCode.Exclude
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/LocalMSALController.java:86]: This may not be necceasary.

**Review by iambmelt — COMMENTED** [review]
So without there being any assignments anywhere that you had to remove as part of your PR, is the assumption that this is some kind of thread-state issue?

**Comment by AdamBJohnsonx** [comment]
> So without there being any assignments anywhere that you had to remove as part of your PR, is the assumption that this is some kind of thread-state issue?

Or I didn't find everything.

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/LocalMSALController.java (hunk 1 above)
Comment (shahzaibj) [reply common/src/main/java/com/microsoft/identity/common/internal/controllers/LocalMSALController.java:86]: Yeah as long as `onlyExplicitlyIncluded` is set to `true`

**Comment by iambmelt** [comment]
@AdamBJohnsonx Do we plan to include all of the various fixes/mitigations we discussed in your PR?

While I'm thinking of it, I'm thinking we'll need to include that `return` statement in `CommandDispatcher` in the below block
```java
    private static BiConsumer<CommandResult, Throwable> getCommandResultConsumer(
            @SuppressWarnings(WarningType.rawtype_warning) @NonNull final BaseCommand command,
            @NonNull final Handler handler) {
        return new BiConsumer<CommandResult, Throwable>() {
            @Override
            public void accept(CommandResult result, final Throwable throwable) {
                if (null != throwable) {
                    handler.post(new Runnable() {
                        @Override
                        public void run() {
                            commandCallBackOnError(command, throwable);
                        }
                    });

                    return; // dont invoke the callback 2x
                }

                // Return command result will post() result for us.
                returnCommandResult(command, result, handler);
            }
        };
    }
```

**Comment by AdamBJohnsonx** [comment]
> @AdamBJohnsonx Do we plan to include all of the various fixes/mitigations we discussed in your PR?
> 
> While I'm thinking of it, I'm thinking we'll need to include that `return` statement in `CommandDispatcher` in the below block
> 
> ```java
>     private static BiConsumer<CommandResult, Throwable> getCommandResultConsumer(
>             @SuppressWarnings(WarningType.rawtype_warning) @NonNull final BaseCommand command,
>             @NonNull final Handler handler) {
>         return new BiConsumer<CommandResult, Throwable>() {
>             @Override
>             public void accept(CommandResult result, final Throwable throwable) {
>                 if (null != throwable) {
>                     handler.post(new Runnable() {
>                         @Override
>                         public void run() {
>                             commandCallBackOnError(command, throwable);
>                         }
>                     });
> 
>                     return; // dont invoke the callback 2x
>                 }
> 
>                 // Return command result will post() result for us.
>                 returnCommandResult(command, result, handler);
>             }
>         };
>     }
> ```

Right.  I made a branch for these fixes, and intended to unmoor it from the other one but did not.

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/LocalMSALController.java (hunk 1 above)
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/LocalMSALController.java:86]:
```suggestion
```

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/LocalMSALController.java [hunk 2]
```diff
@@ -83,9 +83,11 @@
...
     private AuthorizationStrategy mAuthorizationStrategy = null;
 
     @SuppressWarnings(WarningType.rawtype_warning)
+    @EqualsAndHashCode.Exclude
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/LocalMSALController.java:90]:
```suggestion
```

File: common/src/main/java/com/microsoft/identity/common/internal/dto/IdTokenRecord.java [hunk 3]
```diff
@@ -24,6 +24,8 @@
 
 import com.google.gson.annotations.SerializedName;
 
+import lombok.EqualsAndHashCode;
+
 import static com.microsoft.identity.common.internal.dto.IdTokenRecord.SerializedNames.AUTHORITY;
 import static com.microsoft.identity.common.internal.dto.IdTokenRecord.SerializedNames.REALM;
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/dto/IdTokenRecord.java:31]:
```suggestion

@EqualsAndHashCode
```

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerMsalController.java [hunk 4]
```diff
@@ -91,7 +91,10 @@
 
     private static final String TAG = BrokerMsalController.class.getSimpleName();
 
+    @EqualsAndHashCode.Exclude
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerMsalController.java:94]:
```suggestion
```

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerMsalController.java [hunk 5]
```diff
@@ -91,7 +91,10 @@
...
     private BrokerResultFuture mBrokerResultFuture;
+
+    @EqualsAndHashCode.Exclude
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerMsalController.java:97]:
```suggestion
```

**Review by iambmelt — APPROVED** [review]

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/LocalMSALController.java (hunk 1 above)
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/LocalMSALController.java:86]:
```suggestion
```

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/LocalMSALController.java (hunk 2 above)
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/LocalMSALController.java:90]:
```suggestion
```

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerMsalController.java (hunk 4 above)
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerMsalController.java:94]:
```suggestion
```

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerMsalController.java (hunk 5 above)
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerMsalController.java:97]:
```suggestion
```

**Pull request merged**